httpx/client.py

```diff
@@ -147,13 +148,17 @@ def merge_headers(
                 auth = HTTPBasicAuth(username=auth[0], password=auth[1])
             request = auth(request)
 
-        response = await self.send_handling_redirects(
-            request,
-            verify=verify,
-            cert=cert,
-            timeout=timeout,
-            allow_redirects=allow_redirects,
-        )
+        try:
+            response = await self.send_handling_redirects(
+                request,
+                verify=verify,
+                cert=cert,
+                timeout=timeout,
+                allow_redirects=allow_redirects,
+            )
+        except HTTPError as exc:
+            # Add the original request to any HTTPError
+            raise type(exc)(request=request) from None
```

Comment:
Instead of doing this hack why not do:

```python
except HTTPError as exc:
    exc.request = request
    raise
```

Reply:
Indeed in this case it seems (much !) simpler that way, I'll update it